delphi-frontend/src/main/java/au/com/integradev/delphi/preprocessor/DelphiPreprocessor.java

```diff
@@ -119,6 +121,7 @@ public void process() {
 
     tokenStream.fill();
     tokens = extractTokens(tokenStream);
+    rawTokens = List.copyOf(tokens);
```

Comment:
It feels unnecessary to have the tokens stored as a sorted TreeSet, when they end up being stored in an ArrayList anyway.

Retooling `extractTokens` to return a `List<Token>` could have the benefit of slightly more idiomatic code, in which:

- `tokens` is explicitly ordered
- `tokens` is a copy of `rawTokens`, rather than the other way around

```java
private List<Token> tokens;
private List<Token> rawTokens;

// ...

DelphiProcessor(DelphiLexer lexer, DelphiFileConfig config, Platform platform) {
  // ...
  rawTokens = extractTokens(tokenStream);
  tokens = List.copyOf(rawTokens);
  // ...
  tokenStream.setTokens(tokens);
  // ...
}
```

Reply:
How would you handle including new tokens into the `tokens` list via include directives with this approach?

Reply:
Discussed in person - performance of ArrayList<Token> would be worse. Resolving 👍